packages/gatsby-source-mongodb/src/gatsby-node.js

```diff
@@ -148,6 +156,91 @@ function createNodes(
       })
       resolve()
     })
+  }).then(() => {
+    const changeStream = collection.watch()
+
+    changeStream.on(
+      `change`,
+      ({ operationType, documentKey, fullDocument }) => {
+        switch (operationType) {
+          case `replace`:
+          case `insert`: {
+            const id = idToString(documentKey._id)
+            const nodeId = createNodeId(id)
+
+            // if (preserveObjectIds) {
```

Comment:
do we need this?

Reply:
Yeah, we probably should since the first time node creations respect this setting, will dedupe all of this code and include there.